### PR TITLE
Optimize child chain block submission on root chain smart contract

### PR DESCRIPTION
**Pull request by DavidKnott**
Reduces storage used by child chain block submission from 64 bytes to 32 bytes. @kfichter

### CODE REVIEW

File: tests/root_chain/contracts/root_chain/test_root_chain.py [hunk 1]
```diff
@@ -144,9 +150,10 @@ def test_challenge_exit(t, u, root_chain, assert_tx_failed):
     tx_bytes4 = rlp.encode(tx4, UnsignedTransaction)
     merkle = FixedMerkle(16, [tx4.merkle_hash], True)
     proof = merkle.create_membership_proof(tx4.merkle_hash)
+    root = merkle.root
     child_blknum = root_chain.currentChildBlock()
-    root_chain.submitBlock(merkle.root)
-    confirmSig = confirm_tx(tx4, root_chain.getChildChain(child_blknum)[0], key)
+    root_chain.submitBlock(root)
+    confirmSig = confirm_tx(tx4, merkle.root, key)
```
Comment (kfichter) [inline tests/root_chain/contracts/root_chain/test_root_chain.py]: `merkle.root` can also be `root`

File: plasma/root_chain/contracts/RootChain/RootChain.sol [hunk 2]
```diff
@@ -134,19 +124,20 @@ contract RootChain {
         address exitor = txList[6 + 2 * oindex].toAddress(); 
         
         require(msg.sender == exitor);
-        bytes32 root = childChain[blknum].root; 
+        bytes32 root = childChain[blknum];
+        require(root == keccak256(blockRoot, createdAt));
```
Comment (kfichter) [inline plasma/root_chain/contracts/RootChain/RootChain.sol]: Could use `checkRoot` here

File: plasma/root_chain/contracts/RootChain/RootChain.sol [hunk 3]
```diff
@@ -96,34 +88,32 @@ contract RootChain {
...
         uint256 blknum = depositPos / 1000000000;
         // Makes sure that deposit position is actually a deposit
         require(blknum % childBlockInterval != 0);
-        bytes32 root = childChain[blknum].root;
+        bytes32 root = childChain[blknum];
         bytes32 depositHash = keccak256(msg.sender, amount);
-        require(root == depositHash);
-        addExitToQueue(depositPos, msg.sender, amount);
+        bytes32 depositRoot = keccak256(depositHash, createdAt);
+        require(root == depositRoot);
```
Comment (kfichter) [inline plasma/root_chain/contracts/RootChain/RootChain.sol]: Could use `checkRoot` here

**Comment by DavidKnott** [comment]
@kfichter Made the changes you suggested, merging

**Pull request merged**